cmd/influxd/inspect/build_tsi/build_tsi_test.go

```diff
@@ -0,0 +1,423 @@
+	for i := 0; i < 1000; i++ {
+		mms = append(mms, []byte("cpu"))
+		tagSets = append(tagSets, models.NewTags(map[string]string{"region": fmt.Sprintf("r%d", i)}))
+	}
+
+	// Add all to the series file.
+	_, err = sfile.CreateSeriesListIfNotExists(mms, tagSets)
+	require.NoError(t, err)
+
+	params := cmdParams{
+		dataPath:          filepath.Join(tempDir, "data"),
+		walPath:           filepath.Join(tempDir, "wal"),
+		concurrency:       1,
+		compactSeriesFile: true,
+		batchSize:         defaultBatchSize,
+		maxLogFileSize:    tsdb.DefaultMaxIndexLogFileSize,
+		maxCacheSize:      tsdb.DefaultCacheMaxMemorySize,
+	}
+
+	outs := cmdOuts{
+		expectCompactSeries: true,
+		sfile:               sfile,
+	}
+
+	runCommand(t, params, outs)
+}
+
+func initCommand(t *testing.T, params cmdParams) *cobra.Command {
+	t.Helper()
+
+	// Create new command
+	cmd := NewBuildTSICommand()
+
+	// Set args
+	allArgs := make([]string, 0)
+
+	if params.dataPath != os.Getenv("HOME")+"/.influxdbv2/engine/data" {
+		allArgs = append(allArgs, "--data-path", params.dataPath)
+	}
+	if params.walPath != os.Getenv("HOME")+"/.influxdbv2/engine/wal" {
+		allArgs = append(allArgs, "--wal-path", params.walPath)
+	}
+	if params.bucketID != "" {
+		allArgs = append(allArgs, "--bucket-id", params.bucketID)
+	}
+	if params.shardID != "" {
+		allArgs = append(allArgs, "--shard-id", params.shardID)
+	}
+	if params.batchSize != 10000 {
+		allArgs = append(allArgs, "--batch-size", strconv.Itoa(params.batchSize))
+	}
+	if params.maxLogFileSize != tsdb.DefaultMaxIndexLogFileSize {
+		allArgs = append(allArgs, "--max-log-file-size", strconv.Itoa(int(params.maxLogFileSize)))
+	}
+	if params.maxCacheSize != tsdb.DefaultCacheMaxMemorySize {
+		allArgs = append(allArgs, "--max-cache-size", strconv.Itoa(int(params.maxCacheSize)))
+	}
+	if params.compactSeriesFile {
+		allArgs = append(allArgs, "--compact-series-file")
+	}
+	if params.verbose {
+		allArgs = append(allArgs, "--v")
+	}
+	if params.concurrency != runtime.GOMAXPROCS(0) {
+		allArgs = append(allArgs, "--concurrency", strconv.Itoa(params.concurrency))
+	}
+
+	cmd.SetArgs(allArgs)
+
+	return cmd
+}
+
+func getOutput(t *testing.T, cmd *cobra.Command) []byte {
+	t.Helper()
+
+	b := &bytes.Buffer{}
+	cmd.SetOut(b)
+	cmd.SetErr(b)
+	require.NoError(t, cmd.Execute())
+
+	out, err := io.ReadAll(b)
+	require.NoError(t, err)
+
+	return out
+}
+
+func runCommand(t *testing.T, params cmdParams, outs cmdOuts) {
+	t.Helper()
+
+	cmd := initCommand(t, params)
+
+	if outs.expectErr {
+		require.EqualError(t, cmd.Execute(), outs.expectedOut)
+		return
+	}
+
+	if outs.expectBuiltIndex {
+		require.NoDirExists(t, filepath.Join(params.dataPath, "12345", "autogen", "1", "index"))
+		require.NoError(t, cmd.Execute())
+
+		// Check that a valid index directory is present after executing the command
+		isIndex, err := tsi1.IsIndexDir(filepath.Join(params.dataPath, "12345", "autogen", "1", "index"))
+		require.NoError(t, err)
+		require.True(t, isIndex)
```

Comment:
Is there anything we could assert about the contents of the built TSI file, instead of just checking that it exists? Or would it be more trouble than it's worth to open up & read?

Reply:
I agree, I think this would improve the test a lot. Looking into this now, will try to push changes later this morning.

Reply:
Most recent commit improves this a bit. It adds a check of the manifest files, but does not open and read each TSI file. I tried a few things with that, and it didn't seem worth it.